proto-lens/src/Data/ProtoLens/TextFormat.hs

```diff
@@ -203,37 +237,52 @@ modifyField (RepeatedField _ f) value = over f (value :)
+makeValue _ Int64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ UInt32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ UInt64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SInt32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SInt64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ Fixed32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ Fixed64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SFixed32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SFixed64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ FloatField (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ DoubleField (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ BoolField (Parser.IntValue x)
     | x == 0 = Right False
     | x == 1 = Right True
     | otherwise = Left $ "Unrecognized bool value " ++ show x
-makeValue DoubleField (Parser.DoubleValue x) = Right x
-makeValue FloatField (Parser.DoubleValue x) = Right (realToFrac x)
-makeValue BoolField (Parser.EnumValue x)
+makeValue _ DoubleField (Parser.DoubleValue x) = Right x
+makeValue _ FloatField (Parser.DoubleValue x) = Right (realToFrac x)
+makeValue _ BoolField (Parser.EnumValue x)
     | x == "true" = Right True
     | x == "false" = Right False
     | otherwise = Left $ "Unrecognized bool value " ++ show x
-makeValue StringField (Parser.ByteStringValue x) = Right (Text.decodeUtf8 x)
-makeValue BytesField (Parser.ByteStringValue x) = Right x
-makeValue EnumField (Parser.IntValue x) =
+makeValue _ StringField (Parser.ByteStringValue x) = Right (Text.decodeUtf8 x)
+makeValue _ BytesField (Parser.ByteStringValue x) = Right x
+makeValue _ EnumField (Parser.IntValue x) =
     maybe (Left $ "Unrecognized enum value " ++ show x) Right
         (maybeToEnum $ fromInteger x)
-makeValue EnumField (Parser.EnumValue x) =
+makeValue _ EnumField (Parser.EnumValue x) =
     maybe (Left $ "Unrecognized enum value " ++ show x) Right
         (readEnum x)
-makeValue MessageField (Parser.MessageValue x) = buildMessage x
-makeValue GroupField (Parser.MessageValue x) = buildMessage x
-makeValue f val = Left $ "Type mismatch parsing text format: " ++ show (f, val)
+makeValue reg MessageField (Parser.MessageValue Nothing x) =
+  buildMessage reg x
+makeValue reg field@MessageField (Parser.MessageValue (Just typeUri) x)
+    | Just AnyMessageDescriptor { anyTypeUrlLens, anyValueLens } <- matchAnyMessage field =
+        case lookupRegistered typeUri reg of
+          Nothing -> Left "Could not decode Any"
+          Just (SomeMessageType (Proxy :: Proxy value')) ->
+            case buildMessage reg x :: Either String value' of
+              Left err -> Left err
+              Right value' -> Right (def & anyTypeUrlLens .~ typeUri
+                                         & anyValueLens .~ encodeMessage value')
+    | typeUri == messageName (descriptor :: MessageDescriptor value) =
```

Comment:
Remove this case.  The syntax of `[pkg.Foo]{...}` (i.e.without the `type.googleapis.com/` part) refers to proto2 "extensions" which are not yet implemented in proto-lens (#27):
https://github.com/google/protobuf/blob/master/src/google/protobuf/text_format.cc#L402

And in general the type of a proto message is different than the type of an Any that wraps it, so we should avoid confusing the two.